aclk/agent_cloud_link.c

```diff
@@ -954,17 +954,21 @@ static void aclk_main_cleanup(void *ptr)
         // Wakeup thread to cleanup
         QUERY_THREAD_WAKEUP;
         // Send a graceful disconnect message
-        time_t time_created = now_realtime_sec();
         char *msg_id = create_uuid();
 
+        usec_t time_created_offset_usec = now_realtime_usec();
+        time_t time_created = time_created_offset_usec / USEC_PER_SEC;
+        time_created_offset_usec = time_created_offset_usec % USEC_PER_SEC;
```

Comment:
Does using a function instead of repeating these 3 lines of code three times have a performance impact that is not negligible? I don't have any C expertise and I know the agent's code is higly optimized so I'm asking merely out of curiousity.

Reply:
Good point @fracasula  but these 3 lines are used repeatedly only when building a header (with each message that goes to the cloud).

The other two occasions are on MQTT connect to define a LWT and when detecting a graceful shutdown to send a `graceful` message instead of `unexpected` (those two cases are rarely used)